src/game/item_tasks.c

```diff
@@ -0,0 +1,376 @@
+    s->variant = 0;
+    s->unk21 = 0xFF;
+    s->unk1A = 0x200;
+    s->unk1C = 0;
+    s->unk22 = 0x10;
+    s->palId = 0;
+    s->unk10 = SPRITE_FLAG(PRIORITY, 2);
+
+    return t;
+}
+
+struct Task *CreateItemTask_Shield_Magnetic(s8 p0)
+{
+    struct Task *t = TaskCreate(Task_Item_Shield_Magnetic, sizeof(ItemTask), 0x4001, 0,
+                                TaskDestructor_ItemTasks);
+    ItemTask *item = (ItemTask *)TaskGetStructPtr(t);
+
+    item->unk30 = p0;
+
+    {
+        ItemTask *item2 = (ItemTask *)TaskGetStructPtr(t);
+        item2->s.graphics.dest = VramMalloc(36);
+        item2->s.graphics.size = 0;
+        item2->s.graphics.anim = SA2_ANIM_SHIELD_MAGNETIC;
+        item2->s.variant = 0;
+        item2->s.unk21 = 0xFF;
+        item2->s.unk1A = 0x200;
+        item2->s.unk1C = 0;
+        item2->s.unk22 = 0x10;
+        item2->s.palId = 0;
+        item2->s.unk10 = SPRITE_FLAG(PRIORITY, 2);
+    }
+
+    m4aSongNumStart(SE_ACTIVATE_SHIELD);
+
+    return t;
+}
+
+struct Task *CreateItemTask_Confusion(s8 p0)
+{
+    struct Task *t = TaskCreate(Task_Item_Confusion, sizeof(ItemTask), 0x4001, 0,
+                                TaskDestructor_ItemTasks);
+    ItemTask *item = TaskGetStructPtr(t);
+
+    item->unk30 = p0;
+
+    {
+        ItemTask *item2 = (ItemTask *)TaskGetStructPtr(t);
+        item2->s.graphics.dest = VramMalloc(8);
+        item2->s.graphics.size = 0;
+        item2->s.graphics.anim = SA2_ANIM_CONFUSION;
+        item2->s.variant = 0;
+        item2->s.unk21 = 0xFF;
+        item2->s.unk1A = 0x200;
+        item2->s.unk1C = 0;
+        item2->s.unk22 = 0x10;
+        item2->s.palId = 0;
+        item2->s.unk10 = SPRITE_FLAG(PRIORITY, 2);
+    }
+
+    return t;
+}
+
+// https://decomp.me/scratch/DuFBd
+void Task_Item_Shield_Normal(void)
+{
+    s8 param = ITEMTASK_GET_PLAYER_NUM();
+
+    ItemTask *item = TaskGetStructPtr(gCurTask);
+    struct Camera *cam = &gCamera;
+
+    u32 itemEffect
+        = (gPlayer.itemEffect
+           & (PLAYER_ITEM_EFFECT__INVINCIBILITY | PLAYER_ITEM_EFFECT__SHIELD_NORMAL));
+    if (itemEffect != PLAYER_ITEM_EFFECT__SHIELD_NORMAL) {
+        TaskDestroy(gCurTask);
+        return;
+    }
+
+    if (!(gPlayer.itemEffect & PLAYER_ITEM_EFFECT__INVINCIBILITY)) {
+        bool32 b;
+        s32 screenX, screenY;
+
+        screenX = Q_24_8_TO_INT(gPlayer.x) - cam->x;
+        item->s.x = screenX + gPlayer.unk7C;
+
+        screenY = Q_24_8_TO_INT(gPlayer.y) - cam->y;
+        item->s.y = screenY;
+
+        item->s.unk10 &= ~SPRITE_FLAG_MASK_PRIORITY;
+        item->s.unk10 |= gPlayer.unk90->s.unk10 & SPRITE_FLAG_MASK_PRIORITY;
+
+        sub_8004558(&item->s);
+
+#ifndef NON_MATCHING
+        asm("mov %0, %2\n"
+            "and %0, %1\n"
+            : "=r"(b)
+            : "r"(param), "r"(itemEffect));
+
+        // Make the compiler "forget" that itemEffect is 0x1
+        asm("" : "=r"(itemEffect));
+#else
+        b = (param & 0x1);
```

Comment:
Have you tried the `s16 temp = param & 1; b = temp` thing here. Or at least this asm looks similar to the ONE_CYCLE thing

Reply:
Yup, didn't work. :/